src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -565,22 +570,18 @@ impl Coordinator {
         match self.catalog_transact(Some(session), ops).await {
             Ok(()) => {
                 // Determine the initial validity for the table.
-                let since_ts = self.peek_local_write_ts();
+                let register_ts = self.get_local_write_ts().await.timestamp;
 
                 let collection_desc = CollectionDescription::from_desc(
                     table.desc.clone(),
                     DataSourceOther::TableWrites,
                 );
                 self.controller
                     .storage
-                    .create_collections(vec![(table_id, collection_desc)])
+                    .create_collections(register_ts, vec![(table_id, collection_desc)])
                     .await
                     .unwrap_or_terminate("cannot fail to create collections");
-
-                let policy = ReadPolicy::ValidFrom(Antichain::from_elem(since_ts));
-                self.controller
-                    .storage
-                    .set_read_policy(vec![(table_id, policy)]);
+                self.apply_local_write(register_ts).await;
```

Comment:
happy to move it if that's your preference, I can't say I'm confidant yet in my mental model here.

`create_collections` doesn't initialize the upper yet, but it will after the txn stuff is merged. we could potentially add that write manually in the meantime?

Reply:
Okay, as discussed offline, the group commit subsumes this append. So now there's no append to move the `apply` call after, but I did make sure to leave it before the group commit.

Reply:
Update, as discussed offline, went back to append + async group commit so that the newly created table is immediately readable.